numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -162,6 +164,52 @@ PyArray_OptionalIntpConverter(PyObject *obj, PyArray_Dims *seq)
     return PyArray_IntpConverter(obj, seq);
 }
 
+NPY_NO_EXPORT int
+PyArray_CopyConverter(PyObject *obj, PyArray_CopyMode *copymode) {
+    if (obj == Py_None) {
+        PyErr_SetString(PyExc_ValueError,
```

Comment:
```suggestion
        PyErr_SetString(PyExc_TypeError,
```
The error is complaining about an invalid object type here.

Reply:
N.B. - https://github.com/numpy/numpy/pull/19173#discussion_r646740356

I can change the error message from `NoneType ...` to `None`.